ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -905,6 +907,12 @@ private Offset generateReferenceIndexSegmentStartOffset(Offset recordOffset) {
       if (value.isFlagSet(IndexValue.Flags.Delete_Index)) {
         // delete record is always valid
         validEntries.add(new IndexEntry(key, value));
+        if (value.getOriginalMessageOffset() != -1
```

Comment:
Going to be added by @xunyin8 in #554. I have borrowed his code in the meantime.